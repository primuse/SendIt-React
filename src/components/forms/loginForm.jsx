import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { loginUser } from '../../actions/authActions';

class LoginForm extends Component {
  state = {
    email: '',
    password: ''
  };

  componentWillUnmount() {
    // Clear State
    this.setState({
      email: '',
      password: '',
    });
  }

  onChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  };

  login = (event) => {
    event.preventDefault();
    const data = this.state;

    this.props.loginUser(data);
  }

  render() {
    const { email, password } = this.state;
    return <form id="login-form" className="text-center" onSubmit={this.login}>
        <div className="form-row">
          <input type="email" id="email" name="email" value={email}
          placeholder="Email" required onChange={this.onChange} />
        </div>
        <div className="form-row">
          <input type="password" id="pwd" name="password" value={password}
          placeholder="Password" required onChange={this.onChange} />
        </div>
        <button type="submit" className="btn lg is-outlined mb-14">Login</button>
        <div>
          <Link to="/reset">Forgot Password?</Link>
        </div>
      </form>;
  }
}

LoginForm.propTypes = {
  loginUser: PropTypes.func,
};

const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps, { loginUser })(LoginForm);
