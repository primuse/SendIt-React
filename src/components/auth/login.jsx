import React, { Component } from "react";
import { LoginForm, SignupForm } from "../forms/loginForm.jsx";
import { Link } from "react-router-dom";
import "../../css/modules.css";
import "../../css/style.css";
import "../../css/login.css";

export const Login = () => {
  return (
    <section id="login">
      <header>
        <ul>
          <li id="head-brand">
            <Link to="/">
              Send<span className="head-span">IT</span>
            </Link>
          </li>
        </ul>
        <ul className="ml-auto mt-25">
          <li>
            <Link to="/signup">Signup</Link>
          </li>
        </ul>
      </header>
      <div className="banner">
        <div className="sendit-text mb-34">
          <h5>
            The <span className="head-span">Courier</span> service app you've
            been waiting for
          </h5>
          <p>SendIT makes reaching the world easier!</p>
        </div>
        <LoginForm />
      </div>
    </section>
  );
};

export const Signup = () => {
  return (
    <section id="login">
      <header>
        <ul>
          <li id="head-brand">
            <Link to="/">
              Send<span class="head-span">IT</span>
            </Link>
          </li>
        </ul>
        <ul class="ml-auto mt-25">
          <li>
            <Link to="/">Login</Link>
          </li>
        </ul>
      </header>
      <div class="banner">
        <div class="sendit-text">
          <h5>
            The <span class="head-span">Courier</span> service app you've been
            waiting for
          </h5>
          <p class="mb-34">
            Register to experience our world class courier services
          </p>
        </div>
        <div id="form-box">
          <div class="text-center">
            <SignupForm />
          </div>
        </div>
      </div>
    </section>
  );
};
