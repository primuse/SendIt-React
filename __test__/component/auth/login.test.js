import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { Login } from '../../../src/components/auth/login.jsx';
import { LoginForm } from '../../../src/components/forms/loginForm.jsx';

const initialState = {
  auth: {
    isLoading: false
  }
};
const mockStore = configureStore([thunk]);
const store = mockStore(initialState);
const onChange = jest.fn();


describe('Login Page component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});


// describe('Login subscribe form', () => {
//   it('should change state', () => {
//     const wrapper = shallow(<MemoryRouter>
//       <LoginForm />
//     </MemoryRouter>);
//     const input = wrapper.find('#email').dive().instance();
//     console.log(loginForm);
//     // const input = loginForm.find('#email');
//     console.log(input.debug(), 'login===');
//     input.simulate('change', {
//       target: { value: 'yinks@gmail.com' }
//     });
//     // input.instance().value = 'yinks@gmail.com';
//     // input.simulate('change');
//     console.log(loginForm, input.length, loginForm.state());
//     expect(loginForm.state().email).toEqual('yinks@gmail.com');
//   });
// });
