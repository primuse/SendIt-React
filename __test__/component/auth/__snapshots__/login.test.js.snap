// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page component renders correctly 1`] = `
<section
  id="login"
>
  <header>
    <ul>
      <li
        id="head-brand"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Send
          <span
            className="head-span"
          >
            IT
          </span>
        </a>
      </li>
    </ul>
    <ul
      className="ml-auto mt-25"
    >
      <li>
        <a
          href="/signup"
          onClick={[Function]}
        >
          Signup
        </a>
      </li>
    </ul>
  </header>
  <div
    className="banner"
  >
    <div
      className="sendit-text mb-34"
    >
      <h5>
        The 
        <span
          className="head-span"
        >
          Courier
        </span>
         service app you've been waiting for
      </h5>
      <p>
        SendIT makes reaching the world easier!
      </p>
    </div>
    <form
      className="text-center"
      id="login-form"
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="form-row"
      >
        <input
          id="pwd"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn lg is-outlined mb-14"
        type="submit"
      >
        Login
      </button>
      <div>
        <a
          href="/reset"
          onClick={[Function]}
        >
          Forgot Password?
        </a>
      </div>
    </form>
  </div>
</section>
`;
